### PR TITLE
Added general API tests

### PR DESCRIPTION
**Pull request by astrofrog**
Still trying to figure out how to turn this into a markdown table...

**Table of doom:**

Test                 | PolygonPixel | EllipsePixel | RectanglePixel | CirclePixel |
-------------------- | ------------ | ------------ | -------------- | ----------- |
pix_area             | ✔ | ✖ | ✖ | ✖ |
pix_in               | ✔ | ✖ | ✖ | ✖ |
pix_to_mask:all      | ✖ | ✖ | ✖ | ✖ |
pix_to_mask:any      | ✖ | ✖ | ✖ | ✖ |
pix_to_mask:center   | ✖ | ✖ | ✖ | ✖ |
pix_to_mask:exact    | ✖ | ✖ | ✖ | ✖ |
pix_to_shapely       | ✖ | ✖ | ✖ | ✖ |
pix_to_sky:affine    | ✖ | ✖ | ✖ | ✖ |
pix_to_sky:full      | ✖ | ✖ | ✖ | ✖ |
pix_to_sky:local     | ✖ | ✖ | ✖ | ✖ |

Test                 | CircleSky | RectangleSky | EllipseSky | PolygonSky |
-------------------- | --------- | ------------ | ---------- | ---------- |
sky_area             | ✖ | ✖ | ✖ | ✖ |
sky_in               | ✔ | ✖ | ✖ | ✖ |
sky_to_pix:affine    | ✖ | ✖ | ✖ | ✖ |
sky_to_pix:full      | ✖ | ✖ | ✖ | ✖ |
sky_to_pix:local     | ✖ | ✖ | ✖ | ✖ |

### CODE REVIEW

**Comment by cdeil** [comment]
@astrofrog - Can you please rebase this to trigger tests to run again.

Also -- what's the plan here?
Can we merge this now / soon or is this some long-term WIP PR?

**Comment by cdeil** [comment]
@astrofrog Do you have time to finish up this PR in the next two days for the 0.1 release? Or should we punt this one to the 0.2 milestone?

**Comment by astrofrog** [comment]
@cdeil - I've just rebased and now made sure that methods that return ``NotImplementedError`` are xfailed for now.

Note that there *are* some genuine failures, either due to my lack of understanding of the API for certain tests, or real issues with different methods. I don't have time to look at this in the short term, but if someone else could take a look and do a PR to my branch to fix the real failures, then I can merge that in and we can make sure this makes it to 0.1. Some of the failures I saw are definitely of the kind that we should investigate before any release.

**Comment by cdeil** [comment]
@astrofrog - I agree it would be very good to get this in for 0.1. I'll have a look at the fails now and try to fix them.

**Comment by cdeil** [comment]
@astrofrog - Locally this makes all tests pass for me:
https://github.com/astrofrog/regions/pull/1

I guess the main question is if / how `PixCoord` should support array values and indexing, but that should be a separate issue / PR?

One more question: what about the "Table of doom" at the top? Does it make sense to copy it to a separate issue?
Probably we want some kind of such overview table concerning the implementation status for certain regions, right?
But maintaining it as a Markdown table is extra work.
Maybe move it to the wiki where everyone can update it and make a reminder issue?

**Comment by astrofrog** [comment]
> I guess the main question is if / how PixCoord should support array values and indexing, but that should be a separate issue / PR?

IMHO it should do what SkyCoord does behavior wise? (that is, it should be able to represent scalar and vector coordinates). I can't remember if SkyCoord actually stores everything as a numpy array though, I think it does.

**Comment by astrofrog** [comment]
I agree moving the table to the wiki makes sense - can try and do that later today

**Comment by cdeil** [comment]
travis-ci tests passed.
👍  to merge.

File: appveyor.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ environment:
       # For this package-template, we include examples of Cython modules,
       # so Cython is required for testing. If your package does not include
       # Cython code, you can set CONDA_DEPENDENCIES=''
-      CONDA_DEPENDENCIES: "Cython"
+      CONDA_DEPENDENCIES: "Cython shapely"
```
Comment (cdeil) [inline appveyor.yml]: The Appveyor build fails because shapely is not available in this conda channel:
https://ci.appveyor.com/project/Astropy/regions/build/1.0.74/job/wfb95tev3c94p9jh#L81

@astrofrog - Do you know how to fix this? If no, please remove shapely here and let's move the Windows shapely install issue to a separate future issue.

File: regions/core/core.py [hunk 2]
```diff
@@ -46,8 +46,8 @@ def __xor__(self, other):
         return self.symmetric_difference(other)
 
     def __contains__(self, coord):
-        if not coord.isscalar:
-            raise ValueError('{} must be scalar'.format(coord))
+        # if not coord.isscalar:
+        #     raise ValueError('{} must be scalar'.format(coord))
```
Comment (cdeil) [inline regions/core/core.py]: Maybe uncomment this check and implement `PixCoord.isscalar`?
See https://github.com/astropy/regions/issues/25#issuecomment-232321254

**Comment by cdeil** [comment]
@astrofrog - Could you please merge this PR soon?

I'd like to do some more PRs on `regions` and then do the 0.1 release very soon.
We need it for Gammapy, where we have removed `gammapy.regions` and are now using the circle regions from this package, i.e. it's a dependency I'd like to have available before making the Gammapy 0.5 release.

**Comment by astrofrog** [comment]
@cdeil - I'm overloaded today and tomorrow - since you have merge privileges, can you merge this and make the fixes you need to make in a separate PR?

**Comment by astrofrog** [comment]
@cdeil - actually let me push a fix for shapely, can do it now

File: regions/core/core.py (hunk 2 above)
Comment (astrofrog) [inline regions/core/core.py]: I won't have time to do this, can you do it in a separate PR?

**Comment by cdeil** [comment]
Merging this now. Thanks, @astrofrog !

I'll continue with `regions` tomorrow.
